jvm-agent/src/main/org/jetbrains/lincheck/jvm/agent/transformers/LocalVariablesAccessTransformer.kt

```diff
@@ -116,7 +115,42 @@ internal class LocalVariablesAccessTransformer(
     }
 
     private fun getVariableName(varIndex: Int): LocalVariableInfo? {
-        return methodInfo.locals.activeVariables.find { it.index == varIndex }
+        // First, we try to search among active variables.
+        methodInfo.locals.activeVariables.find { it.index == varIndex }?.let { return it }
+
+        // There could be cases where a variable live range given in local variable table entry
```

Comment:
@bbrockbernd suggests sorting events based on line numbers

Also, it might be possible that such cases occur in the case of a local variable "declaration", when the actual initialization (the one related to the source code) happens later. If this is true, we might be able to ignore such "declarations".

Reply:
> @bbrockbernd suggests sorting events based on line numbers

`sum` and `number` initialization events tell the same line number. 

```
sum = Integer;@1 at SimpleProgramNonFailingTest.kt:9
...
numbers = Arrays.ArrayList@1 at SimpleProgramNonFailingTest.kt:9